sympy/polys/tests/test_poly_hypothesis.py

```diff
@@ -0,0 +1,23 @@
+from hypothesis import given
+from hypothesis import assume
+from hypothesis import strategies as st
+from sympy.abc import x
+from sympy.polys.polytools import Poly
+
+
+@given(coefficients1=st.lists(st.integers()), coefficients2=st.lists(st.integers()))
+def test_poly_hypothesis(coefficients1, coefficients2):
+    # non-zero division check.
+    assume(len(coefficients2) * [0] != coefficients2)
```

Comment:
This should be done as a filter on the strategy, like

```
@given(coefficients1=st.lists(st.integers()), 
       coefficients2=st.lists(st.integers()).filter(lambda x: not all(i == 0 for i in x)))
```

(though maybe @honno could suggest a better way too).